### PR TITLE
avoid zero-size VLA and assert file_count to avoid zero division

### PR DESCRIPTION
**Pull request by coderobe**

### CODE REVIEW

File: zindex.c [hunk 1]
```diff
@@ -138,7 +139,6 @@ static void index_tar_entries(ZSTD_DCtx *dctx, const void *base, size_t size, of
             if (is_end_of_archive(tar_data)) {
                 size_t rem = tar_data_size - TAR_BLOCK_SIZE;
                 memmove(tar_data, tar_data + TAR_BLOCK_SIZE, rem);
-                tar_data_size = rem;
```
Comment (alevkov) [inline zindex.c:141]: @coderobe this was a good catch

File: zindex.c [hunk 2]
```diff
@@ -338,9 +338,10 @@ int main(int argc, char **argv) {
     snippet_index idx;
     snippet_index_init(&idx);
 
+    assert(file_count > 0);
```
Comment (alevkov) [inline zindex.c:341]: I assert that `assert` does not belong in production code. @coderobe 

Also,

```

    assert(file_count > 0);
    printf("Just gonna step over this no-op over here, don't mind if I do :3 Now shhhh lets fail silently....");
    int thread_count = (int)(file_count > MAX_THREADS ? MAX_THREADS : file_count);
    pthread_t threads[thread_count];
    thread_arg targs[thread_count];
 ```
 
 ```
 $ gcc -O3 -o zindex zindex.c -I/opt/homebrew/include -L/opt/homebrew/lib -lzstd -DNDEBUG
$ ./zindex "*google*"
Just gonna step over this no-op over here, don't mind if I do :3 Now shhhh lets fail silently....
$ gcc -O3 -o zindex zindex.c -I/opt/homebrew/include -L/opt/homebrew/lib -lzstd
$ ./zindex "*google*"
Assertion failed: (file_count > 0), function main, file zindex.c, line 340.
[1]    73922 abort      ./zindex "*google*"
```

Comment (alevkov) [reply zindex.c:341]: Maybe: @coderobe 

```
if (file_count == 0) {
    fprintf(stderr, "No files found.\n");
    return 1;
}

int thread_count = (file_count > MAX_THREADS) ? MAX_THREADS : file_count;
pthread_t threads[thread_count];
thread_arg targs[thread_count];
```

Comment (coderobe) [reply zindex.c:341]: lgtm

**Pull request merged**